homeassistant/components/kostal_plenticore/const.py

```diff
@@ -0,0 +1,542 @@
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:EnergyPv1:Total",
+        "Energy PV1 Total",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:EnergyPv2:Day",
+        "Energy PV2 Day",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:EnergyPv2:Month",
+        "Energy PV2 Month",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:EnergyPv2:Year",
+        "Energy PV2 Year",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:EnergyPv2:Total",
+        "Energy PV2 Total",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:Yield:Day",
+        "Energy Yield Day",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+            ATTR_ENABLED_DEFAULT: True,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:Yield:Month",
+        "Energy Yield Month",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:Yield:Year",
+        "Energy Yield Year",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+    (
+        "scb:statistic:EnergyFlow",
+        "Statistic:Yield:Total",
+        "Energy Yield Total",
+        {
+            ATTR_UNIT_OF_MEASUREMENT: ENERGY_KILO_WATT_HOUR,
+            ATTR_DEVICE_CLASS: DEVICE_CLASS_ENERGY,
+        },
+        "format_energy",
+    ),
+]
+
+# Defines all entities for settings.
+#
+# Each entry is defined with a tuple of these values:
+#  - module id (str)
+#  - process data id (str)
+#  - entity name suffix (str)
+#  - sensor properties (dict)
+#  - value formatter (str)
+SENSOR_SETTINGS_DATA = [
+    (
+        "devices:local",
+        "Battery:DynamicSoc:Enable",
+        "Battery Dynamic Soc",
```

Comment:
What kind of measurement is this?

Reply:
The Plenticore supports a battery connected to the DC3 string. There seems to be a function which manages the SOC dynamically depending on some previous state. This function can be enabled or disabled and this state can be read with this sensor.

Reply:
If it's a binary sensor it should be part of a binary sensor platform.